### PR TITLE
test: increase coverage build function

### PR DESCRIPTION
**Pull request by AlexD10S**
Unit tests for the build parachain and build contracts processes.

Utilize mocks for the external crates that are used: `duct::cmd` for building parachain and `contract_build::execute` for building contracts.

### CODE REVIEW

File: crates/pop-contracts/src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: GPL-3.0
+use std::path::PathBuf;
+
```
Comment (Daanvdplas) [inline crates/pop-contracts/src/mock.rs:3]: remove empty line

File: crates/pop-contracts/src/mock.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+use contract_build::{
+	BuildResult, ExecuteArgs, MetadataArtifacts, OptimizationResult, OutputType, Verbosity,
+};
```
Comment (Daanvdplas) [inline crates/pop-contracts/src/mock.rs:6]: Add new line

File: crates/pop-contracts/src/lib.rs [hunk 3]
```diff
@@ -8,6 +8,9 @@ mod test;
 mod up;
 mod utils;
```
Comment (Daanvdplas) [inline crates/pop-contracts/src/lib.rs:10]: remove empty line

**Review by Daanvdplas — COMMENTED** [review]
I idea is to perhaps place the parachain tests within this PR together, and the contracts tests. Now there is a split in the tests which can then perhaps be prevented. Not completely sure though

File: crates/pop-cli/src/commands/build/contract.rs [hunk 4]
```diff
@@ -27,3 +28,32 @@ impl BuildContractCommand {
...
+			name: "test_contract".to_string(),
+			path: Some(PathBuf::from(temp_contract_dir.path())),
+		};
+		command.execute().await?;
+
+		Ok(temp_contract_dir)
+	}
+
+	#[tokio::test]
+	async fn test_build_success() -> Result<()> {
```
[the remark this answers is not in the document]
Comment (AlexD10S) [reply crates/pop-cli/src/commands/build/contract.rs:50]: Yes and: 
```
 outro("Build completed successfully!")?;
log::success(result_build.to_string())?;
Ok(())
```
Are you suggestion change the name of the test to make it more clear?

**Pull request closed without merging**